desktop/src/util.rs

```diff
@@ -55,127 +61,276 @@ pub fn winit_to_ruffle_text_control(
+        Key::Named(NamedKey::F9) => LogicalKey::Named(RuffleNamedKey::F9),
+        Key::Named(NamedKey::F10) => LogicalKey::Named(RuffleNamedKey::F10),
+        Key::Named(NamedKey::F11) => LogicalKey::Named(RuffleNamedKey::F11),
+        Key::Named(NamedKey::F12) => LogicalKey::Named(RuffleNamedKey::F12),
+        Key::Named(NamedKey::F13) => LogicalKey::Named(RuffleNamedKey::F13),
+        Key::Named(NamedKey::F14) => LogicalKey::Named(RuffleNamedKey::F14),
+        Key::Named(NamedKey::F15) => LogicalKey::Named(RuffleNamedKey::F15),
+        Key::Named(NamedKey::F16) => LogicalKey::Named(RuffleNamedKey::F16),
+        Key::Named(NamedKey::F17) => LogicalKey::Named(RuffleNamedKey::F17),
+        Key::Named(NamedKey::F18) => LogicalKey::Named(RuffleNamedKey::F18),
+        Key::Named(NamedKey::F19) => LogicalKey::Named(RuffleNamedKey::F19),
+        Key::Named(NamedKey::F20) => LogicalKey::Named(RuffleNamedKey::F20),
+        Key::Named(NamedKey::F21) => LogicalKey::Named(RuffleNamedKey::F21),
+        Key::Named(NamedKey::F22) => LogicalKey::Named(RuffleNamedKey::F22),
+        Key::Named(NamedKey::F23) => LogicalKey::Named(RuffleNamedKey::F23),
+        Key::Named(NamedKey::F24) => LogicalKey::Named(RuffleNamedKey::F24),
+        Key::Named(NamedKey::F25) => LogicalKey::Named(RuffleNamedKey::F25),
+        Key::Named(NamedKey::F26) => LogicalKey::Named(RuffleNamedKey::F26),
+        Key::Named(NamedKey::F27) => LogicalKey::Named(RuffleNamedKey::F27),
+        Key::Named(NamedKey::F28) => LogicalKey::Named(RuffleNamedKey::F28),
+        Key::Named(NamedKey::F29) => LogicalKey::Named(RuffleNamedKey::F29),
+        Key::Named(NamedKey::F30) => LogicalKey::Named(RuffleNamedKey::F30),
+        Key::Named(NamedKey::F31) => LogicalKey::Named(RuffleNamedKey::F31),
+        Key::Named(NamedKey::F32) => LogicalKey::Named(RuffleNamedKey::F32),
+        Key::Named(NamedKey::F33) => LogicalKey::Named(RuffleNamedKey::F33),
+        Key::Named(NamedKey::F34) => LogicalKey::Named(RuffleNamedKey::F34),
+        Key::Named(NamedKey::F35) => LogicalKey::Named(RuffleNamedKey::F35),
+        Key::Character(ch) => {
             // Handle alphabetic characters
-            alpha_to_ruffle_key_code(char).unwrap_or(KeyCode::UNKNOWN)
+            alpha_to_logical(ch)
         }
-        _ => KeyCode::UNKNOWN,
-    };
-    Some(key_code)
+        _ => LogicalKey::Unknown,
+    }
 }
 
-fn alpha_to_ruffle_key_code(char: &str) -> Option<KeyCode> {
-    let char = char.chars().next()?;
-
-    if char.is_ascii_alphabetic() {
-        // ASCII alphabetic characters are all mapped to
-        // their respective KeyCodes, which happen to have
-        // the same numerical value as uppercase characters.
-        return Some(KeyCode::from_code(char.to_ascii_uppercase() as u32));
+fn alpha_to_logical(ch: &str) -> LogicalKey {
+    // TODO What if we get multiple chars?
+    if let Some(ch) = ch.chars().last() {
```

Comment:
Didn't we use the first character before?

Reply:
We weren't consistent:

https://github.com/ruffle-rs/ruffle/pull/19616/files#diff-7be54fe026a0ed3dc2fc1fe118b9028a8c46085c0e5df2a54087572ebbfa9666L219

https://github.com/ruffle-rs/ruffle/pull/19616/files#diff-62c679f6eccebf339181274aa5cd20101388c20fca85d34ccf5ef7e9c016cb77L163

I think getting the last char makes more sense in case of dead keys until we start supporting them properly.